pkg/volume/csi/csi_client.go

```diff
@@ -255,7 +259,10 @@ func (c *csiDriverClient) NodePublishVolume(
 	}
 
 	_, err = nodeClient.NodePublishVolume(ctx, req)
-	return err
+	if err != nil && !isFinalError(err) {
+		return volumetypes.NewUncertainProgressError(err.Error())
+	}
+	return nil
```

Comment:
`return err`!

It took me a while to find out why a pod is running after NodePublish error.

Reply:
fixed. So we had tests for this but fake CSIClient had correct behaviour but real csi client had this bug.. :(